### PR TITLE
Big Split: IPython.kernel

### PR DESCRIPTION
**Pull request by minrk**
There's still some work to do detangling a few of the files (connect, launcher, kernelspec), which have code destined for both packages. But this gets the bulk of the split done, and smaller reorg can happen as more easily reviewed smaller PRs.

### CODE REVIEW

**Comment by minrk** [comment]
Tests are passing, now.

File: ipython_kernel/tests/utils.py [hunk 1]
```diff
@@ -35,7 +36,7 @@ def start_new_kernel(**kwargs):
     
     Integrates with our output capturing for tests.
     """
-    kwargs.update(dict(stdout=nose.iptest_stdstreams_fileno(), stderr=STDOUT))
+    # kwargs.update(dict(stdout=nose.iptest_stdstreams_fileno(), stderr=STDOUT))
```
Comment (Carreau) [inline ipython_kernel/tests/utils.py]: commented ?

File: ipython_kernel/tests/utils.py [hunk 2]
```diff
@@ -123,7 +124,8 @@ def new_kernel(argv=None):
     -------
     kernel_client: connected KernelClient instance
     """
-    kwargs = dict(stdout=nose.iptest_stdstreams_fileno(), stderr=STDOUT,
+    kwargs = dict(
+        # stdout=nose.iptest_stdstreams_fileno(), stderr=STDOUT,
```
Comment (Carreau) [inline ipython_kernel/tests/utils.py]: here also.

**Comment by Carreau** [comment]
Well I suppose the file are mostly just moved... so +1

File: ipython_kernel/tests/utils.py (hunk 1 above)
Comment (minrk) [inline ipython_kernel/tests/utils.py]: Yup. I wanted to run the tests without iptest. I'll do that later. Reverted.

**Comment by Carreau** [comment]
Mergining this.

**Comment by minrk** [comment]
I'd like to land both #7757 and #8107 before merging this.

**Comment by minrk** [comment]
nooooooooooooo, too slow.

**Comment by minrk** [comment]
I'm going to unmerge and reissue this PR.

**Comment by jasongrout** [comment]
Thanks!  I think #7757 is nearly there...

**Comment by Carreau** [comment]
OOops soory

**Comment by Carreau** [comment]
Want to force push master ?

**Comment by minrk** [comment]
@Carreau no problem, I was just in the middle of typing that comment when I saw the merge notification. Force-pushed and reissued as #8175.

**Comment by minrk** [comment]
Actually, there should be no conflicts in #7757, so it's just #8107 to wait for. I thought #7757 was touching comms, but I forgot that PR was merged some time ago.

**Pull request merged**